### ISSUE TITLE
DOI analysis

### ISSUE THREAD

**Issue by priyambial123** [jcrozum/pystablemotifs#83]
I need some help in understanding the domain of influence analysis. For example, when I give a self regulatory node A as 1, the script computes the nodes which will be in OFF (0) and ON (1) states (when node A is ON (1)). Is this right?. Then, script also computes how these nodes in ON and OFF state reach an attractor state. Any node in the network can given as source set or self regulatory nodes are given as source set?. 

Thanks

**Comment by jcrozum** [comment]
The domain of influence code requires you to specify a set of node values to hold fixed. It then tells you which other node values will eventually become fixed as a result. The node values that you specify as being held fixed can be for any nodes, not just autoregulatory nodes.

**Comment by priyambial123** [comment]
Thank you. Is there way to identify the stable motifs in these network. Sorry, if I have missed it, but, I couldn't find it in the Python notebook

**Comment by jcrozum** [comment]
Once you compute the attractor repertoire, e.g., using
```python
ar = sm.AttractorRepertoire.from_primes(primes)
```

You can get a list of the stable motifs that appear in the succession diagram using
```python
motifs = ar.succession_diagram.get_motifs()
```

**Comment by jcrozum** [comment]
In this model, you have 4 explicit source nodes:
```
miR6504*= miR6504
miR93*= miR93
KDM2B*= KDM2B
TFAP2C*= TFAP2C
```
Also, you have 6 variables that do not have update rules specified; these will be treated as source nodes as well.
```
let7a
miR101
miR144
miR744
miR92a
miR3157
```
That gives 10 source nodes, not 3. For 10 source nodes, there are 2^10=1024 possible inputs.
As far as I can tell, there are no positive feedback loops in the network. There is only one feedback loop, and it's a negative feedback loop between MYC and miR98. This will create an oscillation, not a stable motif.

In other words, this network has no stable motifs, and the attractor is determined entirely by the values of the input signals. You have exactly one attractor for each combination of source node values.

**Comment by jcrozum** [comment]
(I should clarify that technically, each source node has two associated "trivial" stable motifs corresponding to each of its two possible values, but these trivial stable motifs are not counted)
